lib/logger/cmd-serializer.ts

```diff
@@ -3,7 +3,7 @@ export default function cmdSerializer(
   cmd: string | string[]
 ): string | string[] {
   if (typeof cmd === 'string') {
-    return cmd.replace(/https:\/\/[^@]*@/g, 'https://**redacted**@'); // TODO #12070
+    return cmd.replace(/https:\/\/[^@]*@/g, 'https://**redacted**@'); // TODO #12070 RE2 causes circular dependency
```

Comment:
We should open an issue upstream for this.

Reply:
This can only be solved by using re2 directly here, right?

Reply:
Why this causes circular dependency? Looks like a re2 bug to me.

Reply:
When re2 is used here it causes the logger class to become undefined in the `lib/logger/index.spec.ts` tests, I have updated the comment to say so.

I will look some more into it to find why  that happens.

Reply:
I understand now, we get a circular module dependency, as regex is using logger and logger is using this.

@rarkins i think we need to stay on normal regex for logger stuff. 🤔